### PR TITLE
Add preinstall script to use latest component library from GitHub

### PR DESCRIPTION
**Pull request by devalevenkatesh**
**Issue #:**
- The meeting demo does not currently get the updated version of react component library and hence may run into build issues.

**Description of changes:**
- This adds a postinstall npm script to clone the react component library so that the meeting demo uses the latest code.

**Testing**

1. How did you test these changes?
- Ran the script locally to check whether the meeting demo can be run with the new changes in Github.
- To do this, installed an older version of react library (2.8.0) and then ran `npm install` which updated the react library to the latest tarball generated with the current code in the react library master branch.
- Did sanity checks and all worked fine.
2. Can these changes be tested using one of the demo application? If yes, which demo application can be used to test it?
- React meeting demo.
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: apps/meeting/package.json [hunk 1]
```diff
@@ -15,11 +15,13 @@
     "react": "^17.0.1",
     "react-dom": "^17.0.1",
     "react-router-dom": "^5.1.2",
+    "rimraf": "^3.0.2",
     "styled-components": "^5.0.0",
     "styled-system": "^5.1.5",
     "throttle-debounce": "^2.3.0"
   },
   "scripts": {
+    "postinstall": "node scripts/update-react-library.js",
```
Comment (ltrung) [inline apps/meeting/package.json:24]: Is there a reason to use postinstall vs preinstall?

File: apps/meeting/scripts/update-react-library.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * GitHub repository.
+ */
+
+const fs = require('fs');
+const path = require('path');
+const { logger, spawnOrFail } = require('./utilities');
+const rimraf = require('rimraf');
+const packageJSON = require('../package.json');
+
+const dir = 'updated-react-library';
```
Comment (ltrung) [inline apps/meeting/scripts/update-react-library.js:14]: Can we call it what it is `amazon-chime-sdk-component-library-react-github`?

File: apps/meeting/scripts/update-react-library.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+if (!fs.existsSync(dir)) {
+  logger.log(`Creating a temp directory ${dir}`);
+  fs.mkdirSync(dir);
+}
+
+process.chdir(path.join(__dirname, `../${dir}`));
+logger.log(`Cloning ${package} to update the existing installation.`);
+spawnOrFail('git', [`clone ${packageGitUrl}`]);
+logger.log('Cloning complete.');
```
Comment (ltrung) [inline apps/meeting/scripts/update-react-library.js:27]: nit: Dont think we need this line.

File: apps/meeting/scripts/update-react-library.js (hunk 2 above)
Comment (ltrung) [reply apps/meeting/scripts/update-react-library.js:14]: or github packages if we want to maybe do js sdk in the future :D

File: apps/meeting/package.json (hunk 1 above)
Comment (devalevenkatesh) [reply apps/meeting/package.json:24]: Any works, just wanted to do it once the rest are installed. Do not think it really matters here, should I use pre instead?

File: apps/meeting/scripts/update-react-library.js (hunk 2 above)
Comment (devalevenkatesh) [reply apps/meeting/scripts/update-react-library.js:14]: Yeah makes sense, will update it thanks!

File: apps/meeting/scripts/update-react-library.js (hunk 3 above)
Comment (devalevenkatesh) [reply apps/meeting/scripts/update-react-library.js:27]: Ok, removed it 👍

File: apps/meeting/package.json (hunk 1 above)
Comment (michhyun1) [reply apps/meeting/package.json:24]: If I ever want to use the npm version, how would I do that?

If I do npm install amazon-chime-sdk-component-library-react@2.9.0

wouldnt the postinstall script run and just overwrite the amazon-chime-sdk-component-library-react dependency to point to github master?

Comment (devalevenkatesh) [reply apps/meeting/package.json:24]: Yeah good question and I have tested it already. So, the `post` `pre` install scripts will only run on the `npm install` script and not `npm install <package>`.

**Pull request closed without merging**